### PR TITLE
Devx 935

### PR DESCRIPTION
**Pull request by MonicaG**
Updates for mobile developer guide

### CODE REVIEW

**Comment by MonicaG** [comment]
The updated documentation for this PR can be reviewed at https://dev.developer.gov.bc.ca/docs/default/component/mobile-developer-guide/distribution_methods/

File: docs/distribution_methods.md [hunk 1]
```diff
@@ -18,15 +18,75 @@ Employee apps are distributed through [MDM inTune by the OCIO Device Management
...
+
+Custom apps go through Apple's app review process. The same app review guidelines for App Store Apps apply to Custom Apps.
+
+Custom Apps are licensed from [Apple Business Manager (ABM)](https://support.apple.com/en-ca/guide/apple-business-manager/welcome/web) and assigned to the OCIO Device Management team's Intune Tenancy. They will distribute the app through MDM-Intune. 
+
+#### Product Owner responsibilities
+
+The Product Owner will work with the relevant teams to decide whether to use the OCIO’s ABM or the Ministry’s own ABM.
+
+If the Ministry decides to use its own ABM, it would need to create and manage it. As part of the setup it would add the OCIO’s MDM-Intune Organization ID as a location.
```
Comment (jenreiher-bcgov) [inline docs/distribution_methods.md:33]: do we have a recommended path here? It seems like folks prefer us to indicate the "paved road"

File: docs/assets/distribution.drawio.svg
Comment (jenreiher-bcgov) [inline docs/assets/distribution.drawio.svg:1]: I see you're adding the enterprise program in here - which is fine, but perhaps we can highlight the ABM as "recommended"?

File: docs/distribution_methods.md [hunk 2]
```diff
@@ -18,15 +18,75 @@ Employee apps are distributed through [MDM inTune by the OCIO Device Management
...
+#### Developer resposibilities
+
+The developer handles developing the app and providing the binary to the Product Owner. A repeatable CI/CD pipeline is expected as part of this process.
+
+
+#### OCIO Device Management team resposibilities
+
+The OCIO Device Management Team will distribute the app to employee devices. They will work with the Product Owner to schedule the release.
+
+### Enterprise vs Custom App
```
Comment (jenreiher-bcgov) [inline docs/distribution_methods.md:68]: let's put this above the other two - then folks can jump down to the one they're interested in if this helps them make the decision up from

File: docs/distribution_methods.md [hunk 3]
```diff
@@ -18,15 +18,75 @@ Employee apps are distributed through [MDM inTune by the OCIO Device Management
...
+
+
+#### OCIO Device Management team responsibilities
+
+The OCIO Device Management Team will distribute the app to employee devices. They will work with the Product Owner to schedule the release.
+
+
+### Apple Enterprise App
+
+Use this distribution method for internal apps used only by employees. These apps are specifically made for the Province and are not SaaS solutions. This distribution method is for apps that can't go through Apple's app review process.
```
Comment (jenreiher-bcgov) [inline docs/distribution_methods.md:50]: It sounds like we want to discourage new applications from using this, so perhaps let's add something here to that effect? (if I'm remembering our conversations correctly?)

File: docs/distribution_methods.md [hunk 4]
```diff
@@ -18,15 +18,75 @@ Employee apps are distributed through [MDM inTune by the OCIO Device Management
...
+#### OCIO Device Management team resposibilities
+
+The OCIO Device Management Team will distribute the app to employee devices. They will work with the Product Owner to schedule the release.
+
+### Enterprise vs Custom App
+
+|    | Enterprise  App | Custom App |
+| -- | --------------- | ---------- |
+| App goes through Apple's App Review | No | Yes |
+| App usable with expired certificate |  No |  Yes |
```
Comment (jenreiher-bcgov) [inline docs/distribution_methods.md:73]: it doesn't feel like some folks understood the implications of an expired certificate, so we might need to add context somehow that makes it clear the impact of this

File: docs/distribution_methods.md [hunk 5]
```diff
@@ -18,15 +18,75 @@ Employee apps are distributed through [MDM inTune by the OCIO Device Management
...
+
+The OCIO Device Management Team will distribute the app to employee devices. They will work with the Product Owner to schedule the release.
+
+### Enterprise vs Custom App
+
+|    | Enterprise  App | Custom App |
+| -- | --------------- | ---------- |
+| App goes through Apple's App Review | No | Yes |
+| App usable with expired certificate |  No |  Yes | 
+| App usable with expired provisioning profile |  No |  Yes |
```
Comment (jenreiher-bcgov) [inline docs/distribution_methods.md:74]: same here

File: docs/distribution_methods.md [hunk 6]
```diff
@@ -18,15 +18,75 @@ Employee apps are distributed through [MDM inTune by the OCIO Device Management
...
+
+### Enterprise vs Custom App
+
+|    | Enterprise  App | Custom App |
+| -- | --------------- | ---------- |
+| App goes through Apple's App Review | No | Yes |
+| App usable with expired certificate |  No |  Yes | 
+| App usable with expired provisioning profile |  No |  Yes | 
+| Can be a SaaS solution | No | Yes |
+| TestFlight | No | Yes |
```
Comment (jenreiher-bcgov) [inline docs/distribution_methods.md:76]: Can be released via TestFlight

File: docs/distribution_methods.md [hunk 7]
```diff
@@ -18,15 +18,75 @@ Employee apps are distributed through [MDM inTune by the OCIO Device Management
...
+### Apple Unlisted App
+
+Use this method when your app is an internal app used by employees and contractors. Refer to [Apple's documentation](https://developer.apple.com/support/unlisted-app-distribution/) on how to distribute unlisted apps. These apps require some extra setup in App Store Connect. The [Developer Experience team](contact.md) will assist in this setup.
+
+Unlisted apps will go through Apple's app review process. Ensure your app meets [Apple's guidelines](https://developer.apple.com/app-store/review/).
 
 ## App review process
 Both the Apple App Store and Google Play Store have an app review process. 
 
 When creating your app, it's important to consider how the app reviewer will interact with your app. For example, if your app requires a user to login, the app reviewer will need either an account or access to a demo account.
```
Comment (jenreiher-bcgov) [inline docs/distribution_methods.md:88]: or the team can submit a recording of the app?

File: docs/distribution_methods.md (hunk 1 above)
Comment (MonicaG) [reply docs/distribution_methods.md]: I've updated to include a recommended path

File: docs/distribution_methods.md (hunk 2 above)
Comment (MonicaG) [reply docs/distribution_methods.md]: Moved the table to above.

File: docs/distribution_methods.md (hunk 4 above)
Comment (MonicaG) [reply docs/distribution_methods.md]: Added section about implications of expiring assets

File: docs/distribution_methods.md (hunk 6 above)
Comment (MonicaG) [reply docs/distribution_methods.md]: Renamed to "Beta Test with TestFlight"

File: docs/distribution_methods.md (hunk 7 above)
Comment (MonicaG) [reply docs/distribution_methods.md:95]: BC Services Card is a special case. I don't want to put recording of the app into the docs and have teams plan to use it and get rejected.

**Review by jenreiher-bcgov — APPROVED** [review]
This is looking good to me
I think there is some assumed knowledge about the pros and cons of these different choices that only a developer would understand HOWEVER that is the audience for this documentation.

I will add a ticket to the backlog to work with Kaitlyn's team on a version of this to go into digital.gov that cross-links to this that unpacks the PO decision points and gives more opinionated advice that they can benefit from.

**Review by CoffeeGurlNia — COMMENTED** [review]
I've committed a couple, small content edits.

File: docs/distribution_methods.md [hunk 8]
```diff
@@ -18,19 +18,75 @@ Employee apps are distributed through [MDM inTune by the OCIO Device Management
 
 [Contact us](contact.md) at the start of your project so we can discuss the process for distributing an internal app. It requires setup from both the Developer Experience and OCIO Device Management teams.
 
-Apps used by both employees and contractors also need extra setup in Apple's App Store Connect. The [Developer Experience Team](contact.md) will assist in this setup.
-
 ### Apple Custom App 
 
+Use this distribution method for internal apps used only by employees. These apps are either made specifically for the Province of BC or are a SaaS solution.
```
Comment (CoffeeGurlNia) [inline docs/distribution_methods.md]: Province of **B.C.**

**Review by CoffeeGurlNia — APPROVED** [review]

File: docs/distribution_methods.md [hunk 9]
```diff
@@ -2,31 +2,106 @@
...
+    
+    The Product Owner must schedule yearly updates to these assets, so they don't expire.
+
+Custom and Public apps do not have this limitation. Your app will work with expired assets. 
+
+The intellectual property (IP) of a SaaS app belongs to the app's developer. The developer uses their provisioning profile and signing certificate. They are responsible for updating these assets.
+
+### Apple Custom App 
+
+Use this distribution method for internal apps used only by employees. These apps are either made specifically for the Province of BC or are a SaaS solution not available on the public app store. The IP of purpose-built apps belongs to the Province of BC. The IP of SaaS apps belong to the developer.
```
Comment (MonicaG) [inline docs/distribution_methods.md:48]:
```suggestion
Use this distribution method for internal apps used only by employees. These apps are either made specifically for the Province of B.C. or are a SaaS solution not available on the public app store. The IP of purpose-built apps belongs to the Province of BC. The IP of SaaS apps belong to the developer. 
```

File: docs/distribution_methods.md [hunk 10]
```diff
@@ -2,31 +2,106 @@
...
+![Diagram of the various distribution methods. If the app is for the general public, it's distributed via the Apple App Store and Google Play Stores. If it's an app for Government employees, it's distributed via MDM inTunes. If it's an app for Employees and Contractors, it is distributed as an unlisted app via the Apple App Store.](assets/distribution.drawio.svg)
 
 ## Public apps
-Public apps are released through the Province's accounts on the Apple App Store and Google Play Store. 
+Public apps are released through the Province's accounts on the Apple App Store and Google Play Store. These apps are used by the general public.
 
-You must list your app through the Province's accounts. The [Developer Experience team](contact.md) helps you get your app on these accounts.
+You must list your app through the Province's accounts. The [Developer Experience team](contact.md) will help you get your app on these accounts.
+
+[Contact us](contact.md) if your app's intellectual property (IP) doesn't belong to the Province of BC. We will work with you to determine how to release your app.
```
Comment (MonicaG) [inline docs/distribution_methods.md:12]:
```suggestion
[Contact us](contact.md) if your app's intellectual property (IP) doesn't belong to the Province of B.C. We will work with you to determine how to release your app.
```

File: docs/distribution_methods.md [hunk 11]
```diff
@@ -2,31 +2,106 @@
...
+### Apple Custom App vs Enterprise App
+
+|    | Apple Custom  App | Apple Enterprise App |
+| -- | --------------- | ---------- |
+| App goes through Apple's App Review | Yes | No |
+| App usable with expired signing certificate |  Yes |  No | 
+| App usable with expired provisioning profile |  Yes |  No | 
+| Beta test with [TestFlight](https://developer.apple.com/testflight/) | Yes | No |
+| Can be a SaaS solution | Yes | No |
+| Who owns the app's IP | Province of BC or Developer | Province of BC |
```
Comment (MonicaG) [inline docs/distribution_methods.md:33]:
```suggestion
| Who owns the app's IP | Province of BC or Developer | Province of B.C. |
```

Comment (MonicaG) [inline docs/distribution_methods.md:33]:
```suggestion
| Who owns the app's IP | Province of B.C. or Developer | Province of BC |
```

File: docs/distribution_methods.md (hunk 9 above)
Comment (MonicaG) [inline docs/distribution_methods.md:48]:
```suggestion
Use this distribution method for internal apps used only by employees. These apps are either made specifically for the Province of BC or are a SaaS solution not available on the public app store. The IP of purpose-built apps belongs to the Province of B.C. The IP of SaaS apps belong to the developer. 
```

File: docs/distribution_methods.md [hunk 12]
```diff
@@ -2,31 +2,106 @@
...
+
+
+#### OCIO Device Management team responsibilities
+
+The OCIO Device Management Team will distribute the app to employee devices. They will work with the Product Owner to schedule the release.
+
+
+### Apple Enterprise App
+
+Use this distribution method for internal apps used only by employees. These apps are specifically made for the Province and are not SaaS solutions. The Province of BC holds the IP of these apps. They are signed with the Province's signing certificate.
```
Comment (MonicaG) [inline docs/distribution_methods.md:73]:
```suggestion
Use this distribution method for internal apps used only by employees. These apps are specifically made for the Province and are not SaaS solutions. The Province of B.C. holds the IP of these apps. They are signed with the Province's signing certificate.
```

File: docs/distribution_methods.md (hunk 10 above)
Comment (MonicaG) [inline docs/distribution_methods.md:12]:
```suggestion
[Contact us](contact.md) if your app's intellectual property (IP) doesn't belong to the Province of B.C. We will work with you to determine how to release your app.
```

File: docs/distribution_methods.md (hunk 11 above)
Comment (MonicaG) [inline docs/distribution_methods.md:33]:
```suggestion
| Who owns the app's IP | Province of B.C. or Developer | Province of B.C. |
```

File: docs/distribution_methods.md (hunk 9 above)
Comment (MonicaG) [inline docs/distribution_methods.md:48]:
```suggestion
Use this distribution method for internal apps used only by employees. These apps are either made specifically for the Province of B.C. or are a SaaS solution not available on the public app store. The IP of purpose-built apps belongs to the Province of B.C. The IP of SaaS apps belong to the developer. 
```

File: docs/distribution_methods.md (hunk 12 above)
Comment (MonicaG) [inline docs/distribution_methods.md:73]:
```suggestion
Use this distribution method for internal apps used only by employees. These apps are specifically made for the Province and are not SaaS solutions. The Province of B.C. holds the IP of these apps. They are signed with the Province's signing certificate.
```

**Pull request merged**